### PR TITLE
Cats and corgis will no longer mate with kittens and puppies.

### PR DESCRIPTION
**Pull request by harmonyn**
Fixes https://github.com/tgstation/tgstation/issues/27550. There are a number of ways to do this, I went with MOB_SIZE_SMALL since kittens/puppies are small and regular animals aren't (only other mob that uses this is gutlunch, which is never small).

### CODE REVIEW

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 1]
```diff
@@ -346,7 +346,9 @@
 	for(var/mob/M in view(7, src))
 		if(M.stat != CONSCIOUS) //Check if it's conscious FIRST.
 			continue
-		else if(istype(M, childtype)) //Check for children SECOND.
+		if(M.mob_size == MOB_SIZE_SMALL) // Check if it's underage SECOND
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: you broke the else if chain, not that it matters much since it continues

**Comment by WJohn** [comment]
there's so many things wrong with this i dont even know where to start

good job fixing it

**Comment by Iamgoofball** [comment]
please hold off on this I need to refactor for dwarves anyways

**Comment by bgobandit** [comment]
@WJohn can you be more specifci

**Comment by MrPerson** [comment]
Excuse me goof why aren't you up in arms about this LITERAL BEASTIALITY

**Comment by YakumoChen** [comment]
:-1: ruins catsplosion, would not feature

**Comment by Iamgoofball** [comment]
This is stupid yes but I need to rework this code to fix catsplosion not working the way I want it AND making sure cats grow up

**Pull request closed without merging**